api/src/cleanup/cleanup.service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2025 Hexastack. All rights reserved.
+ *
+ * Licensed under the GNU Affero General Public License v3.0 (AGPLv3) with the following additional terms:
+ * 1. The name "Hexabot" is a trademark of Hexastack. You may not use this name in derivative works without express written permission.
+ * 2. All derivative works must include clear attribution to the original creator and software, Hexastack and Hexabot, in a prominent location (e.g., in the software's "About" section, documentation, and README file).
+ */
+
+import { Injectable } from '@nestjs/common';
+
+import { ChannelService } from '@/channel/channel.service';
+import { HelperService } from '@/helper/helper.service';
+import { LoggerService } from '@/logger/logger.service';
+import { SettingService } from '@/setting/services/setting.service';
+import { DeleteResult } from '@/utils/generics/base-repository';
+
+import { TCriteria, TExtractExtension, TExtractNamespace } from './types';
+
+@Injectable()
+export class CleanupService {
+  constructor(
+    private readonly helperService: HelperService,
+    private readonly loggerService: LoggerService,
+    private readonly settingService: SettingService,
+    private readonly channelService: ChannelService,
+  ) {}
+
+  private async deleteMany(criteria: TCriteria[]): Promise<DeleteResult> {
```

Comment:
```suggestion
  private async deleteManyBySuffixAndNamespaces(criteria: TCriteria[]): Promise<DeleteResult> {
```
I suggest this function should live inside ethe setting service.